packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/stage-parser.spec.ts

```diff
@@ -0,0 +1,379 @@
+        isValid: true,
+      },
+      {
+        usecase: 'stage with scaler value with comments',
+        expression: `{$count: 20 //limits doc count\n }`,
+        operator: '$count',
+        value: `20 //limits doc count`,
+        isValid: true,
+      },
+
+      {
+        usecase: 'object with more then one stage key',
+        expression: `{$match: {name: /berlin/i}, $count: 20}`,
+        operator: '',
+        isValid: false,
+      },
+      {
+        usecase: 'stage name without $',
+        expression: `{match: {name: /berlin/i}, }`,
+        operator: '',
+        isValid: false,
+      },
+    ];
+
+    it('sets node disabled prop', function () {
+      const node = babelParser.parseExpression(`{$match: {name: /berlin/i}}`);
+
+      setNodeDisabled(node, true);
+      expect(isNodeDisabled(node)).to.be.true;
+
+      setNodeDisabled(node, false);
+      expect(isNodeDisabled(node)).to.be.false;
+    });
+
+    it('asserts stage node', function () {
+      const data = [
+        {
+          expression: `[]`,
+          errorMessage: 'Each element of the pipeline array must be an object',
+        },
+        {
+          expression: `'name'`,
+          errorMessage: 'Each element of the pipeline array must be an object',
+        },
+        {
+          expression: `{}`,
+          errorMessage: 'A pipeline stage specification object must contain exactly one field.',
+        },
+        {
+          expression: `{match: {}}`,
+          errorMessage: 'Stage value can not be empty',
+        },
+      ];
+      data.forEach(({ expression, errorMessage }) => {
+        try {
+          assertStageNode(babelParser.parseExpression(expression))
+        } catch (e) {
+          expect(e).to.be.instanceOf(SyntaxError);
+          expect((e as SyntaxError).message).to.equal(errorMessage);
+        }
+      });
+
+      stages.filter(x => x.isValid).forEach(({ expression }) => {
+        assertStageNode(babelParser.parseExpression(expression));
+      });
+    });
+
+    it('stage operator from node', function () {
+      stages.filter(x => x.isValid).forEach(({ expression, usecase, operator }) => {
+        expect(getStageOperatorFromNode(
+          babelParser.parseExpression(expression) as any),
+          usecase
+        ).to.equal(operator);
+      });
+    });
+
+    it('stage value from node', function () {
+      stages.filter(x => x.isValid).forEach(({ expression, usecase, value }) => {
+        expect(getStageValueFromNode(
+          babelParser.parseExpression(expression) as any),
+          usecase
+        ).to.equal(value);
+      });
+    });
+
+    it('converts stage to comment', function () {
+      const stage = babelParser.parseExpression(`{$match: {name: /berlin/i}}`);
+      const comments = stageToComments(stage);
+      comments.forEach(({ type }) => expect(type).to.equal('CommentLine'));
+      const value = comments.map(({ value }) => value).join('\n');
+      expect(value).to.equal([
```

Comment:
well honestly i don't have any strong preference. 
in other places i changed it to string, but here i'll leave it as is because of the extra space we add when converting to line comments, which i find easier to write/read.